### ISSUE TITLE
[Bug]: PHP8.1 - Implicit conversion from float to int loses precision at /3rdparty/deepdiver/zipstreamer/src/Count64.php#121

### ISSUE THREAD

**Issue by solracsf** [nextcloud/server#35171]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

See error log.

### Steps to reproduce

N/A

### Expected behavior

No error.

### PHP engine version

PHP 8.1

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
N/A
```

### Nextcloud Logs

```shell
{
  "reqId": "gYeENZ4vo5RyfgkrWykn",
  "level": 3,
  "time": "2022-11-15T08:17:37+00:00",
  "remoteAddr": "5.50.56.174",
  "user": "--",
  "app": "PHP",
  "method": "GET",
  "url": "/s/Rx6bg48H8nRLnQj/download",
  "message": "Implicit conversion from float 1.8446744069414584E+19 to int loses precision at /var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/Count64.php#121",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 Edg/107.0.1418.42",
  "version": "24.0.7.1",
  "exception": {
    "Exception": "Error",
    "Message": "Implicit conversion from float 1.8446744069414584E+19 to int loses precision at /var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/Count64.php#121",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/Count64.php",
        "line": 121,
        "function": "onError",
        "class": "OC\\Log\\ErrorHandler",
        "type": "::",
        "args": [
          8192,
          "Implicit conversion from float 1.8446744069414584E+19 to int loses precision",
          "/var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/Count64.php",
          121
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/ZipStreamer.php",
        "line": 387,
        "function": "ZipStreamer\\pack64le",
        "args": [
          {
            "__class__": "ZipStreamer\\Lib\\Count64_64"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/ZipStreamer.php",
        "line": 480,
        "function": "buildZip64ExtendedInformationField",
        "class": "ZipStreamer\\ZipStreamer",
        "type": "->",
        "args": [
          {
            "__class__": "ZipStreamer\\Lib\\Count64_64"
          },
          {
            "__class__": "ZipStreamer\\Lib\\Count64_64"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/deepdiver/zipstreamer/src/ZipStreamer.php",
        "line": 259,
        "function": "buildCentralDirectoryHeader",
        "class": "ZipStreamer\\ZipStreamer",
        "type": "->",
        "args": [
          "Fichier candidats/",
          null,
          0,
          0,
          {
            "__class__": "ZipStreamer\\Lib\\Count64_64"
          },
          {
            "__class__": "ZipStreamer\\Lib\\Count64_64"
          },
          0,
          1106051088,
          true
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/Streamer.php",
        "line": 115,
        "function": "addEmptyDir",
        "class": "ZipStreamer\\ZipStreamer",
        "type": "->",
        "args": [
          "Fichier candidats/"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/legacy/OC_Files.php",
        "line": 215,
        "function": "addDirRecursive",
        "class": "OC\\Streamer",
        "type": "->",
        "args": [
          "/INTERWAY/Fichier candidats"
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/files_sharing/lib/Controller/ShareController.php",
        "line": 710,
        "function": "get",
        "class": "OC_Files",
        "type": "::",
        "args": [
          "/INTERWAY",
          "Fichier candidats",
          [
            false
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 225,
        "function": "downloadShare",
        "class": "OCA\\Files_Sharing\\Controller\\ShareController",
        "type": "->",
        "args": [
          "Rx6bg48H8nRLnQj",
          null,
          "",
          ""
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 133,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Files_Sharing\\Controller\\ShareController"
          },
          "downloadShare"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 172,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Files_Sharing\\Controller\\ShareController"
          },
          "downloadShare"
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/private/Route/Router.php",
        "line": 298,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Files_Sharing\\Controller\\ShareController",
          "downloadShare",
          {
            "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
          },
          [
            "",
            "Rx6bg48H8nRLnQj",
            "files_sharing.Share.downloadShare"
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/lib/base.php",
        "line": 1030,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/s/Rx6bg48H8nRLnQj/download"
        ]
      },
      {
        "file": "/var/www/nextcloud/index.php",
        "line": 36,
        "function": "handleRequest",
        "class": "OC",
        "type": "::",
        "args": []
      }
    ],
    "File": "/var/www/nextcloud/lib/private/Log/ErrorHandler.php",
    "Line": 92,
    "CustomMessage": "--"
  },
  "id": "63734bdcd9b8b"
}
```

**Comment by szaimen** [comment]
cc @come-nc

**Comment by szaimen** [comment]
Hi, please update to 25.0.7 or better 26.0.2 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 26-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by solracsf** [comment]
Not fixed on 25.0.7 RC1

**Comment by solracsf** [comment]
Fixed by https://github.com/DeepDiver1975/PHPZipStreamer/pull/11
